docs/tutorials/on-device-training/ios-app.md

```diff
@@ -0,0 +1,987 @@
+
+    we also have `progressCallback` that will be called after each training step. We will use this callback to update the progress bar in the UI.
+
+    ```swift
+    private let kNumOtherRecordings: Int = 20
+    private let kNumEpochs: Int = 3
+    
+    let kUserIndex: Int64 = 1
+    let kOtherIndex: Int64 = 0
+
+    func train(_ trainingData: [Data], progressCallback: @escaping (Double) -> Void) throws {
+        let numRecordings = trainingData.count
+        var otherRecordings = Array(0..<kNumOtherRecordings)
+        for e in 0..<kNumEpochs {
+            print("Epoch: \(e)")
+            otherRecordings.shuffle()
+            let otherData = otherRecordings.prefix(numRecordings)
+            
+            for i in 0..<numRecordings {
+                let (buffer, wavFileData) = try getDataFromWavFile(fileName: "other_\(otherData[i])")
+                try trainStep(inputData: [trainingData[i], wavFileData], labels: [kUserIndex, kOtherIndex])
+                print("finished training on recording \(i)")
+                
+                let progress = Double((e * numRecordings) + i + 1) / Double(kNumEpochs * numRecordings)
+                progressCallback(progress)
+            }
+        }
+        
+    }
+    ```
+
+3.  #### Exporting the trained model
+
+    We can use the `exportModelForInference` method of the `ORTTrainingSession` class to export the trained model. The method takes in the path where the model should be exported and the output names of the model.
+    
+    Here we will export the model to the `Library` directory of the application. The exported model will be used for inference purposes.
+
+    ```swift
+    func exportModelForInference() throws {
+        guard let libraryDirectory = FileManager.default.urls(for: .libraryDirectory, in: .userDomainMask).first else {
+            throw TrainerError.Error("Failed to find library directory ")
+        }
+        
+        let modelPath = libraryDirectory.appendingPathComponent("inference_model.onnx").path
+        try trainingSession.exportModelForInference(withOutputPath: modelPath, graphOutputNames: ["output"])
+    }
+    ```
+
+You can find the complete implementation of the `Trainer` class [here](https://github.com/microsoft/onnxruntime-training-examples/blob/master/on_device_training/mobile/ios/MyVoice/Trainer.swift)
+
+
+### Inference with the trained model
+The `VoiceIdentifier` class will handle the inference with the trained model. It will load the trained model and perform inference on the given audio data. The class will have the `evaluate(inputData: Data) -> Result<(Bool, Float), Error>` method that will take in the audio data and return the result of the inference. The result will be a tuple of `(Bool, Float)`, where the first element represents whether the audio belongs to the user or not, and the second element represents the confidence score of the prediction.
```

Comment:
```suggestion
The `VoiceIdentifier` class will handle the inference with the trained model. It will load the trained model and perform inference on the given audio data. The class will have the `evaluate(inputData: Data) -> Result<(Bool, Float), Error>` method that will take in the audio data and return the result of the inference. The result will be a tuple of `(Bool, Float)`, where the first element represents whether the audio is identified to be the user or not, and the second element represents the confidence score of the prediction.
```